src/main/java/engine/movegen/MovesGenerator.java

```diff
@@ -0,0 +1,50 @@
+package engine.movegen;
+
+import java.util.List;
+
+import engine.move.Move;
+import engine.entities.Board;
+import engine.entities.Piece;
+
+/**
+ * Abstract class containing useful methods used in
+ * its subclasses. Each subclass will generate a
+ * list of possible moves according to the piece
+ * it's related to, and will then call this method
+ * removeInvalid to remove moves not on board or
+ * occupied by friendly pieces.
+ */
+public abstract class MovesGenerator {
+    public List<Move> moves;
+    public abstract List<Move> generate (Board board, Piece piece);
+    /**
+     * Mutates list of moves by removing moves not on
+     * board or occupied by friendly pieces.
+     */
+    public void removeInvalid (List<Move> moves, Board board) {
+        // TODO: will use checkTile method
+    }
+    /**
+     * Returns true if the destination of move is on
+     * board and is NOT occupied by friendly piece.
+     */
+    public boolean checkTile (Move move, Board board) {
+        // TODO: will use methods containsAbsCoords from board class and isOccupiedByFriendly and
+        return true;
```

Comment:
Consider adding the following error to incomplete methods:
`throw new java.lang.UnsupportedOperationException("Not implemented yet.");`

as opposed to simply returning `true`

Reply:
We will do from now on whenever writing an incomplete method.